### PR TITLE
Finish email validator export

### PR DESCRIPTION
**Pull request by jcholder-legal**
Fixed `modules.adonis` so that it can build properly again.

Upgraded the `formattedEmail` validation function to a full-on validation `@Directive`.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,16 +1,17 @@
 import { NgModule } from '@angular/core';
 import { SplitPhonePipe } from './pipes/split_phone.pipe';
 import { PhoneInputFormatter } from './directives/phone_input_formatter.directive';
-import { EmailValidators } from './validators';
-import { onlyLetters, formattedDate, minDate } from './validators';
+import { FormattedEmailValidator } from './validators/email.validators';
+import { onlyLetters, formattedDate, minDate } from './validators/index';
```
Comment (kylessnell) [inline src/index.ts]: shouldn't have to do this?

File: src/index.ts [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
 
 export * from './directives/phone_input_formatter.directive';
 export * from './pipes/split_phone.pipe';
-export * from './validators';
+export * from './validators/email.validators';
+export * from './validators/index';
```
Comment (kylessnell) [inline src/index.ts]: node looks for an `index` file by default

File: src/index.ts (hunk 1 above)
Comment (jcholder-legal) [reply src/index.ts]: You wouldn't think but that's _exactly_ what was breaking the build.

File: src/index.ts (hunk 2 above)
Comment (jcholder-legal) [reply src/index.ts]: Same as above: The build breaks if we don't do this.

File: src/validators/email.validators.spec.ts [hunk 3]
```diff
@@ -1,9 +1,64 @@
...
+    it('returns an error in order to report that the input is not valid', () => {
+      el.value = 'example_example.com';
+
+      expect(directiveInstance.validate(el)).toEqual({ formattedEmail: { valid: false } });
+    });
+  });
+});
 
-describe('EmailValidators', () => {
+/*
```
Comment (kylessnell) [inline src/validators/email.validators.spec.ts]: remove comments?

File: src/validators/email.validators.ts [hunk 4]
```diff
@@ -1,13 +1,32 @@
...
-  public static formattedEmail(): ValidatorFn {
-    return (abstractControl: AbstractControl): { [key: string]: any } => {
-      const emailRegex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
-      let email = abstractControl.value
-      let validEmail = !email.length || email.match(emailRegex);
+const emailRegex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
 
-      return validEmail ? null : { 'email': { value: abstractControl.value } };
-    };
+export function formattedEmail() {
```
Comment (kylessnell) [inline src/validators/email.validators.ts]: why export this?

File: src/validators/email.validators.spec.ts (hunk 3 above)
Comment (jcholder-legal) [reply src/validators/email.validators.spec.ts]: Oops! I had meant to re-integrate those specs after the upgrade.

File: src/validators/email.validators.ts (hunk 4 above)
Comment (jcholder-legal) [reply src/validators/email.validators.ts]: Leftover from testing. Removing.

File: src/index.ts (hunk 2 above)
Comment (cryptojuice) [reply src/index.ts]: I agree with Kyle, something seems fishy about having to do this.

Comment (jcholder-legal) [reply src/index.ts]: True, but something I want to save for the upcoming pre-flight build check changes. I don't want to hold up the `onboarding.adonis` changes in order to chase this down.

**Pull request merged**